multichain-testing/test/deposit-withdraw-lca.test.ts

```diff
@@ -0,0 +1,222 @@
+import anyTest from '@endo/ses-ava/prepare-endo.js';
+import type { TestFn } from 'ava';
+import { AmountMath } from '@agoric/ertp';
+import { makeDoOffer } from '../tools/e2e-tools.js';
+import { makeQueryClient } from '../tools/query.js';
+import { commonSetup, SetupContextWithWallets } from './support.js';
+
+const test = anyTest as TestFn<SetupContextWithWallets>;
+
+const accounts = ['user1'];
+
+const contractName = 'basicFlows';
+const contractBuilder =
+  '../packages/builders/scripts/orchestration/init-basic-flows.js';
+
+test.before(async t => {
+  const { deleteTestKeys, setupTestKeys, ...rest } = await commonSetup(t);
+  deleteTestKeys(accounts).catch();
+  const wallets = await setupTestKeys(accounts);
```

Comment:
Aside...

In https://github.com/Agoric/dapp-orchestration-basics/pull/61 where I folded in [your work on starship testing](https://github.com/Agoric/dapp-orchestration-basics/pull/50), I struggled a bit to get this `commonSetup(t)` stuff working. In order to be able to test it without re-deploying my contract for each iteration (or worse: standing up the whole cluster again), I factored out the ambient authority so I could inject mocks for quick ava testing.
 
See `test('orca-multichain.test style usage', ...)`.
https://github.com/Agoric/dapp-orchestration-basics/blob/63f85352eaa73659d4cadbc1f17d48dd4095d98c/contract/test/agd-lib.test.js#L234

and `test('deploy-cli style usage', ...)`
https://github.com/Agoric/dapp-orchestration-basics/blob/main/contract/test/agd-lib.test.js#L214

Here's hoping we can migrate that back to `support.js` here in due course.

Reply:
Nice, that's a good approach for this problem. We added `startContact` in https://github.com/Agoric/agoric-sdk/pull/9864/commits/3639831175485d5500965a5e493b007683fe8dbf a bit ago that ensures the contract is only installed once by querying vstorage

I wonder what the best path is for keeping testing tools in sync with minimal effort. Releases were something I'd floated in the past but it didn't seem to get much support.